src/site/layouts/home.drupal.liquid

```diff
@@ -1,11 +1,11 @@
 {% include "src/site/includes/header.html" with drupalTags = true %}
-<main data-template="layouts/home">
+
+<div data-widget-type="authenticated-homepage"></div>
+<main data-template="layouts/home" id="homepage">
```

Comment:
Just a thought: what if this `main` was hidden by default and a new div that contained a spinner was shown. Then the new React code would control the visibility of the spinner div, static content, and new logged-in homepage content.

Reply:
That's actually what Nick was suggesting with making the homepage a full on react app I think -- the problem is that all users will definitely see a spinner at first even when they aren't logged in, as opposed to now where the homepage just loads immediately 🙂

Reply:
Ah... Wasn't Nick also talking about feeding data from CMS into a new React version of the signed-out page?

Yeah, there are definite tradeoffs either way. The weird flash of static logged-out content every single time you go to the home page while signed in will also not be great.

I am also starting to come around to the idea of redoing the logged-out home page in React. I could see how there would be a wrapper component that decides which homepage to show, and our team would own the logged-in version and your team (maybe?) would own the new logged-out version.

Reply:
> Ah... Wasn't Nick also talking about feeding data from CMS into a new React version of the signed-out page?

Definitely - we would modify `home.drupal.liquid` to write the CMS data as JSON into the HTML as something like `window.VetsGov.homepage`. We do this with the megamenu. Then we would just grab the reference in the React component.

> Yeah, there are definite tradeoffs either way. The weird flash of static logged-out content every single time you go to the home page while signed in will also not be great.

You could add some inline JavaScript directly into the `home.drupal.liquid` to check synchronously for a session variable. If it exists, hide the unauth exp, show the spinner, and try to load the user's data. Otherwise, synchronously show the unauth page.

> I am also starting to come around to the idea of redoing the logged-out home page in React. I could see how there would be a wrapper component that decides which homepage to show, and our team would own the logged-in version and your team (maybe?) would own the new logged-out version.

That's totally fine by me. It would be much cleaner technically and the homepage bundle could be lighter weight than the static-pages bundle, so it might even load faster.